json_schema_for_humans/generate.py

```diff
@@ -328,7 +345,8 @@ def generate_from_file_object(
     """Generate the JSON schema documentation from opened file objects for both input and output files. The
     result_file should be opened in write mode.
     """
-    result = generate_from_schema(json.load(schema_file), minify, deprecated_from_description, default_from_description)
+    schema_path = os.path.abspath(schema_file.name).split(os.path.sep)
```

Comment:
Unfortunately that does not work on Windows because `os.path.sep` is `\`, but the references in the schemas are always with forward slashes. I tested the following:
```suggestion
    schema_path = list(Path(schema_file.name).resolve().parts)
```
Don't forget to add the import `from pathlib import Path` at line 5. I cannot make that a suggestion, because it is too far from any code modified in the current PR.

Reply:
Agreed that all references in schemas should be forward slashes.  Inside of `generate_from_file_object` and `generate_from_filename` we should be dealing with the schema file passed via the command line (or test), which should have an OS-specific path (and `os.path.abspath(...)` will always return a path delimited by `os.path.sep`).  I ran it on Windows (cmd and powershell (and Linux for a sanity check)) and confirmed it worked as expected.  That said, I'm fine switching to pathlib if you'd like.  The only gotcha I see is that we need schema_path to be an absolute path; otherwise if schemas have refs with more '../'s than the relative path specified, we won't be able to resolve the reference.

Reply:
You are right that it works, there was something wrong with my test.

I'm not sure I understand the last part of your message correctly, but the `.resolve()` part makes the path absolute.

It's probably best to keep it the way you did at first as it is simpler.

Reply:
Ah ok, thanks. If `.resolve()` produces an absolute path then I think either solution would work.